formal-models/dbft3.0/dbft.tla

```diff
@@ -0,0 +1,430 @@
+  \* set means that all nodes are good in every possible behaviour.
+  \* Examples: {0}
+  \*           {1, 3}
+  \*           {}
+  RMFault,
+
+  \* RMDead is a set of consensus node indexes that are allowed to die in the
+  \* middle of every behaviour and do not send any message afterwards. RMDead
+  \* must be a subset of RM. An empty set means that all nodes are alive and
+  \* responding in in every possible behaviour. RMDead may intersect the
+  \* RMFault set which means that node which is in both RMDead and RMFault
+  \* may become FAULT and send any message starting from some step of the
+  \* particular behaviour and may also die in the same behaviour which will
+  \* prevent it from sending any message.
+  \* Examples: {0}
+  \*           {3, 2}
+  \*           {}
+  RMDead,
+
+  \* MaxView is the maximum allowed view to be considered (starting from 0,
+  \* including the MaxView itself). This constraint was introduced to reduce
+  \* the number of possible model states to be checked. It is recommended to
+  \* keep this setting not too high (< N is highly recommended).
+  \* Example: 2
+  MaxView
+
+VARIABLES
+  \* rmState is a set of consensus node states. It is represented by the
+  \* mapping (function) with domain RM and range RMStates. I.e. rmState[r] is
+  \* the state of the r-th consensus node at the current step.
+  rmState,
+
+ \* msgs is the shared pool of messages sent to the network by consensus nodes.
+ \* It is represented by a subset of Messages set.
+  msgs
+
+\* vars is a tuple of all variables used in the specification. It is needed to
+\* simplify fairness conditions definition.
+vars == <<rmState, msgs>>
+
+\* N is the number of validators.
+N == Cardinality(RM)
+
+\* F is the number of validators that are allowed to be malicious.
+F == (N - 1) \div 3
+
+\* M is the number of validators that must function correctly.
+M == N - F
+
+\* These assumptions are checked by the TLC model checker once at the start of
+\* the model checking process. All the input data (declared constants) specified
+\* in the "Model Overview" section must satisfy these constraints.
+ASSUME
+  /\ RM \subseteq Nat
+  /\ N >= 4
+  /\ 0 \in RM
+  /\ RMFault \subseteq RM
+  /\ RMDead \subseteq RM
+  /\ Cardinality(RMFault) <= F
+  /\ Cardinality(RMDead) <= F
+  /\ Cardinality(RMFault \cup RMDead) <= F
+  /\ MaxView \in Nat
+  /\ MaxView <= 2
+
+\* RMStates is a set of records where each record holds the node state and
+\* the node current view.
+RMStates == [
+              type: {"initialized", "prepareSent", "commitSent", "cv", "ackSent", "blockAccepted", "bad", "dead"},
+              view : Nat
+            ]
+
+\* Messages is a set of records where each record holds the message type,
+\* the message sender and sender's view by the moment when message was sent.
+Messages == [type : {"PrepareRequest", "PrepareResponse", "Commit", "Ack", "ChangeView"}, rm : RM, view : Nat]
```

Comment:
Do we have any name for this new message in the paper? `Ack` is not that bad, but doesn't sound perfect either.

Reply:
I think the only one I have on paper is from the recent docs:
```
1. New consensus period - The second commit period;
2. New consensus payload - Decryption sharing message;
```
How about `postCommit` and `postCommitSent`?

Reply:
Actually, dBFT 3.0 has `Pre-Commit-P1` and `Pre-Commit-P2` which may be sent if the node receives f+1 preparations. These messages should be followed by `Commit` then. But I doubt that we may use `preCommit` name in the model since it tights us to the dBFT 3.0 meaning. So how about `postCommit`?

Reply:
dBFT 3.0 is largely irrelevant for this (except we don't want to collide with it). There is a concept of "PreBlock" in some docs as well. "CommitBlock"? "CommitAck"? "AcceptBlock"? "CommitUpdate"? "UpdatedCommit"? It's a commit, but with an additional data that was not available at the time of the first commit.

Reply:
Let's take the `CommitAck`.